components/testldap/slapd-init.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+set -eu
+
+reconfigure_slapd() {
+    echo "Reconfigure slapd..."
+    cat <<EOL | debconf-set-selections
+slapd slapd/internal/generated_adminpw password ${LDAP_SECRET}
+slapd slapd/internal/adminpw password ${LDAP_SECRET}
+slapd slapd/password2 password ${LDAP_SECRET}
+slapd slapd/password1 password ${LDAP_SECRET}
+slapd slapd/dump_database_destdir string /var/backups/slapd-VERSION
+slapd slapd/domain string ${LDAP_DOMAIN}
+slapd shared/organization string ${LDAP_ORGANISATION}
+slapd slapd/backend string MDB
+slapd slapd/purge_database boolean true
+slapd slapd/move_old_database boolean true
+slapd slapd/allow_ldap_v2 boolean false
+slapd slapd/no_configuration boolean false
+slapd slapd/dump_database select when needed
+EOL
+
+    DEBIAN_FRONTEND=noninteractive dpkg-reconfigure slapd
+}
+
+
+make_snakeoil_certificate() {
+    echo "Make snakeoil certificate for ${LDAP_DOMAIN}..."
+    openssl req -subj "/CN=${LDAP_DOMAIN}" \
+                -new \
+                -newkey rsa:2048 \
+                -days 365 \
```

Comment:
Does this happen on every container start or only on container build? Perhaps 10y is nice to avoid a surprise next year :D